File: Webgui/web/js/model/taak.js
define(['jquery',
        'knockout',
        'underscore'],
    function ($, ko, _) {

        return function taak() {
            var _this = this;

            _this.identificatie = ko.observable();
            _this.tijdstipCreatie = ko.observable();
            _this.deadline = ko.observable();
            _this.tijdstipAfgehandeld = ko.observable();
            _this.titel = ko.observable();
            _this.omschrijving = ko.observable();
            _this.entiteitId = ko.observable();
            _this.soortEntiteit = ko.observable();
            _this.wijzigingTaaks = ko.observableArray([]);
            _this.toegewezenAan = ko.observable();
            _this.status = ko.observable();

            _this.berekenToegewezenAanEnStatus = function() {
                var wt = _.chain(_this.wijzigingTaaks())
                .sortBy('tijdstip')
                .filter(function(wijzigingTaak) {
                    return wijzigingTaak.toegewezenAan() != null;
                })
//                .map(function(wijzigingTaak) {
//                    return wijzigingTaak.toegewezenAan();
//                })
                .last()
                .value();

                if(wt != null){
                    _this.toegewezenAan(wt.toegewezenAan());

                    var status = wt.taakStatus();

                    _this.status(status.substring(0, 1).toUpperCase() + status.substring(1).toLowerCase());
                }
            }

        };
    });

File: Clients/src/main/java/nl/lakedigital/djfc/commons/domain/response/Taak.java
package nl.lakedigital.djfc.commons.domain.response;

import java.util.List;

import static com.google.common.collect.Lists.newArrayList;

public class Taak {
    private String identificatie;
    private String tijdstipCreatie;
    private String deadline;
    private String tijdstipAfgehandeld;
    private String titel;
    private String omschrijving;
    private Long entiteitId;
    private String soortEntiteit;
    private String toegewezenAan;
    private List<WijzigingTaak> wijzigingTaaks;

    public String getIdentificatie() {
        return identificatie;
    }

    public void setIdentificatie(String identificatie) {
        this.identificatie = identificatie;
    }

    public String getTijdstipCreatie() {
        return tijdstipCreatie;
    }

    public void setTijdstipCreatie(String tijdstipCreatie) {
        this.tijdstipCreatie = tijdstipCreatie;
    }

    public String getDeadline() {
        return deadline;
    }

    public void setDeadline(String deadline) {
        this.deadline = deadline;
    }

    public String getTijdstipAfgehandeld() {
        return tijdstipAfgehandeld;
    }

    public void setTijdstipAfgehandeld(String tijdstipAfgehandeld) {
        this.tijdstipAfgehandeld = tijdstipAfgehandeld;
    }

    public String getTitel() {
        return titel;
    }

    public void setTitel(String titel) {
        this.titel = titel;
    }

    public String getOmschrijving() {
        return omschrijving;
    }

    public void setOmschrijving(String omschrijving) {
        this.omschrijving = omschrijving;
    }

    public Long getEntiteitId() {
        return entiteitId;
    }

    public void setEntiteitId(Long entiteitId) {
        this.entiteitId = entiteitId;
    }

    public String getSoortEntiteit() {
        return soortEntiteit;
    }

    public void setSoortEntiteit(String soortEntiteit) {
        this.soortEntiteit = soortEntiteit;
    }

    public String getToegewezenAan() {
        return toegewezenAan;
    }

    public void setToegewezenAan(String toegewezenAan) {
        this.toegewezenAan = toegewezenAan;
    }

    public List<WijzigingTaak> getWijzigingTaaks() {
        if (wijzigingTaaks == null) {
            wijzigingTaaks = newArrayList();
        }
        return wijzigingTaaks;
    }

    public void setWijzigingTaaks(List<WijzigingTaak> wijzigingTaaks) {
        this.wijzigingTaaks = wijzigingTaaks;
    }
}
